Mathlib/Topology/Homeomorph.lean

```diff
@@ -49,139 +49,140 @@ infixl:25 " ≃ₜ " => Homeomorph
 
-instance : CoeFun (α ≃ₜ β) fun _ ↦ α → β := ⟨FunLike.coe⟩
+instance : CoeFun (X ≃ₜ Y) fun _ ↦ X → Y := ⟨FunLike.coe⟩
 
 @[simp]
-theorem homeomorph_mk_coe (a : Equiv α β) (b c) : (Homeomorph.mk a b c : α → β) = a :=
+theorem homeomorph_mk_coe (a : Equiv X Y) (b c) : (Homeomorph.mk a b c : X → Y) = a :=
   rfl
 #align homeomorph.homeomorph_mk_coe Homeomorph.homeomorph_mk_coe
 
 /-- Inverse of a homeomorphism. -/
-protected def symm (h : α ≃ₜ β) : β ≃ₜ α where
+protected def symm (h : X ≃ₜ Y) : Y ≃ₜ X where
   continuous_toFun := h.continuous_invFun
   continuous_invFun := h.continuous_toFun
   toEquiv := h.toEquiv.symm
 #align homeomorph.symm Homeomorph.symm
 
-@[simp] theorem symm_symm (h : α ≃ₜ β) : h.symm.symm = h := rfl
+@[simp] theorem symm_symm (h : X ≃ₜ Y) : h.symm.symm = h := rfl
 #align homeomorph.symm_symm Homeomorph.symm_symm
 
 /-- See Note [custom simps projection] -/
-def Simps.symm_apply (h : α ≃ₜ β) : β → α :=
+def Simps.symm_apply (h : X ≃ₜ Y) : Y → X :=
   h.symm
 #align homeomorph.simps.symm_apply Homeomorph.Simps.symm_apply
 
 initialize_simps_projections Homeomorph (toFun → apply, invFun → symm_apply)
 
 @[simp]
-theorem coe_toEquiv (h : α ≃ₜ β) : ⇑h.toEquiv = h :=
+theorem coe_toEquiv (h : X ≃ₜ Y) : ⇑h.toEquiv = h :=
   rfl
 #align homeomorph.coe_to_equiv Homeomorph.coe_toEquiv
 
 @[simp]
-theorem coe_symm_toEquiv (h : α ≃ₜ β) : ⇑h.toEquiv.symm = h.symm :=
+theorem coe_symm_toEquiv (h : X ≃ₜ Y) : ⇑h.toEquiv.symm = h.symm :=
   rfl
 #align homeomorph.coe_symm_to_equiv Homeomorph.coe_symm_toEquiv
 
 @[ext]
-theorem ext {h h' : α ≃ₜ β} (H : ∀ x, h x = h' x) : h = h' :=
+theorem ext {h h' : X ≃ₜ Y} (H : ∀ x, h x = h' x) : h = h' :=
   FunLike.ext _ _ H
 #align homeomorph.ext Homeomorph.ext
 
 /-- Identity map as a homeomorphism. -/
 @[simps! (config := { fullyApplied := false }) apply]
-protected def refl (α : Type*) [TopologicalSpace α] : α ≃ₜ α where
+protected def refl (X : Type*) [TopologicalSpace X] : X ≃ₜ X where
   continuous_toFun := continuous_id
   continuous_invFun := continuous_id
-  toEquiv := Equiv.refl α
+  toEquiv := Equiv.refl X
 #align homeomorph.refl Homeomorph.refl
 
 /-- Composition of two homeomorphisms. -/
-protected def trans (h₁ : α ≃ₜ β) (h₂ : β ≃ₜ γ) : α ≃ₜ γ where
+protected def trans (h₁ : X ≃ₜ Y) (h₂ : Y ≃ₜ Z) : X ≃ₜ Z where
   continuous_toFun := h₂.continuous_toFun.comp h₁.continuous_toFun
   continuous_invFun := h₁.continuous_invFun.comp h₂.continuous_invFun
   toEquiv := Equiv.trans h₁.toEquiv h₂.toEquiv
 #align homeomorph.trans Homeomorph.trans
 
 @[simp]
-theorem trans_apply (h₁ : α ≃ₜ β) (h₂ : β ≃ₜ γ) (a : α) : h₁.trans h₂ a = h₂ (h₁ a) :=
+theorem trans_apply (h₁ : X ≃ₜ Y) (h₂ : Y ≃ₜ Z) (a : X) : h₁.trans h₂ a = h₂ (h₁ a) :=
   rfl
 #align homeomorph.trans_apply Homeomorph.trans_apply
 
-@[simp] theorem symm_trans_apply (f : α ≃ₜ β) (g : β ≃ₜ γ) (a : γ) :
+@[simp] theorem symm_trans_apply (f : X ≃ₜ Y) (g : Y ≃ₜ Z) (a : Z) :
```

Comment:
Here the original variable `a` didn't correspond to its type `γ`, but `z : Z` is probably better.